### PR TITLE
Should care about ScanWithIndex in ScalarDbUtils.copyAndSetTargetToIfNot()

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/util/ScalarDbUtils.java [hunk 1]
```diff
@@ -67,6 +67,8 @@ public static Scan copyAndSetTargetToIfNot(
     Scan ret;
     if (scan instanceof ScanAll) {
       ret = new ScanAll((ScanAll) scan); // copy
+    } else if (scan instanceof ScanWithIndex) {
+      ret = new ScanWithIndex((ScanWithIndex) scan); // copy
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/util/ScalarDbUtils.java:71]: I realized that we missed handling `ScanWithIndex` here.

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**